doc/devdocs/modules/keyboardmanager/keyboardmanager.md

```diff
@@ -0,0 +1,199 @@
+- The profile format (`default.json`) is of the following format:
+
+        {
+            "remapKeys":
+            {
+                "inProcess":
+                [
+                    {
+                        "originalKeys":"91",
+                        "newRemapKeys":"162;70"
+                    },
+                    {
+                        "originalKeys":"92",
+                        "newRemapKeys":"162;70"
+                    }
+                ]
+            },
+            "remapShortcuts":
+            {
+                "global":
+                [
+                    {
+                        "originalKeys":"164;37",
+                        "newRemapKeys":"162;65"
+                    },
+                    {
+                        "originalKeys":"162;68",
+                        "newRemapKeys":"91"
+                    }
+                ],
+                "appSpecific":
+                [
+                    {
+                        "originalKeys":"91;162;65",
+                        "newRemapKeys":"162;86",
+                        "targetApp":"msedge"
+                    }
+                ]
+            }
+        }
+
+    - `originalKeys` stores the key/shortcut which is to be pressed for the remap, and `newKeys` stores the key/shortcut which is to be executed.
+    - Both contain semi-colon separated virtual key codes. For `remapKeys`, `originalKeys` must have only one key code, whereas for `remapShortcuts` it must have atleast two key codes.
+    - `inProcess` sub-key was added in `remapKeys` because there was a possibility of adding the registry based remapping approach (used by [SharpKeys](https://github.com/randyrants/sharpkeys)), so that would be under a separate sub-key while `inProcess` would be for keyboard hook based remaps. This was deprioritized as there weren't enough requests for it.
+    - `remapShortcuts` is split into `global` and `appSpecific`, where `global` remaps would apply to all applications, whereas `appSpecific` would apply on when the `targetApp` is in focus. `targetApp` must be the process name of the app (with or without it's extension), e.g. `msedge` or `msedge.exe` for Microsoft Edge.
+
+## Loading settings
+KBM settings are loaded only on the C++ side only at start up, in the [constructor](https://github.com/microsoft/PowerToys/blob/b80578b1b9a4b24c9945bddac33c771204280107/src/modules/keyboardmanager/dll/dllmain.cpp#L67-L68). The settings file may get modified from the KBM UI on applying new remappings, but the the file is not read again. The files are read from the PowerToys Settings process whenever a change is made to the file (using a FileWatcher) or whenever the KBM page is opened.
```

Comment:
Would be good to have the communication part more documented. For example:
-  When do in-memory data structures(osLevelShortcutReMapSortedKeys, appSpecificShortcutReMapSortedKeys...) get updated? 
- Links to code 

If it is described somewhere else the link should be enough.

Reply:
I have add a link here to a section in the keyboardmanagerui.md file, and I have updated that section with more information and code links on the in-memory data structures.